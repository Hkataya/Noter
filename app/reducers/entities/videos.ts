/* eslint-disable @typescript-eslint/no-unused-vars */
import { Action } from 'redux';
import { allvideos } from '../../normalized-state';

export default function videos(state = allvideos, _action: Action<string>) {
  return state;
}
