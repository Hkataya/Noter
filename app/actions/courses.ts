import { CourseType } from '../reducers/entities/types';
import { Dispatch } from '../reducers/types';
import CourseRepository from '../db/RepositoryInitializer';

export const ADD_COURSE = 'ADD_COURSE';
export const REMOVE_COURSE = 'REMOVE_COURSE';
export const UPDATE_COURSE = 'UPDATE_COURSE';
export const FETCH_ALL_COURSES = 'FETCH_ALL_COURSES';

type AddCourseAction = {
  type: typeof ADD_COURSE;
  payload: CourseType;
};

type RemoveCourseAction = {
  type: typeof REMOVE_COURSE;
  payload: CourseType['id'];
};

type UpdateCourseAction = {
  type: typeof UPDATE_COURSE;
  payload: CourseType;
};

type FetchAllCoursesAction = {
  type: typeof FETCH_ALL_COURSES;
  payload: {
    courses: Array<CourseType>;
  };
};

export type CourseActionCreatorType = {
  addCourseDb?: (courseData: Omit<CourseType, 'id'>) => unknown;
  removeCourseDb?: (courseId: CourseType['id']) => unknown;
  updateCourseDb?: (courseData: CourseType) => unknown;
  fetchAllCoursesDb?: () => unknown;
};

export type CourseActionType =
  | AddCourseAction
  | RemoveCourseAction
  | UpdateCourseAction
  | FetchAllCoursesAction;

export function addCourse(courseData: CourseType) {
  return {
    type: ADD_COURSE,
    payload: courseData
  };
}

export function removeCourse(courseId: CourseType['id']) {
  return {
    type: REMOVE_COURSE,
    payload: courseId
  };
}

export function updateCourse(courseData: CourseType) {
  return {
    type: UPDATE_COURSE,
    payload: courseData
  };
}

export function fetchAllCourses(coursesData: Array<CourseType>) {
  return {
    type: FETCH_ALL_COURSES,
    payload: {
      courses: coursesData
    }
  };
}

export function addCourseDb(courseData: Omit<CourseType, 'id'>) {
  return (dispatch: Dispatch) => {
    CourseRepository.createEntity(courseData)
      .then(updatedData =>
        dispatch(addCourse(Object.assign(courseData, updatedData)))
      )
      .catch(err => {
        console.log(err);
      });
  };
}

export function removeCourseDb(courseId: CourseType['id']) {
  return (dispatch: Dispatch) => {
    CourseRepository.deleteEntity(courseId)
      .then(() => dispatch(removeCourse(courseId)))
      .catch(err => {
        console.log(err);
      });
  };
}
export function fetchAllCoursesDb() {
  return (dispatch: Dispatch) => {
    CourseRepository.getAllCourses()
      .then((courses: Array<CourseType>) => {
        return dispatch(fetchAllCourses(courses));
      })
      .catch(err => {
        console.log(err);
      });
  };
}
export function updateCourseDb(courseData: CourseType) {
  return (dispatch: Dispatch) => {
    CourseRepository.updateEntity(courseData)
      .then(() => dispatch(updateCourse(courseData)))
      .catch(err => {
        console.log(err);
      });
  };
}
