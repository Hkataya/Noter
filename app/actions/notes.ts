import { VideoType, NoteType } from '../reducers/entities/types';

export const ADD_NOTE = 'ADD_NOTE';
export const REMOVE_NOTE = 'REMOVE_NOTE';

type AddNoteAction = {
  type: typeof ADD_NOTE;
  payload: {
    noteId: NoteType['id'];
    videoId: VideoType['id'];
    noteData: NoteType;
  };
};

type RemoveNoteAction = {
  type: typeof REMOVE_NOTE;
  payload: {
    videoId: VideoType['id'];
    noteId: NoteType['id'];
  };
};

export type NoteActionCreatorType = {
  addNote?: (noteData: NoteType, videoId: VideoType['id']) => void;
  removeNote?: (noteId: NoteType['id'], courseId: VideoType['id']) => void;
};

export type NoteActionType = AddNoteAction | RemoveNoteAction;

export function addNote(noteData: NoteType, videoId: VideoType['id']) {
  // Generate Unique ID
  const noteId = Date.now().toString();
  const note = {
    id: noteId,
    ...noteData
  };
  return {
    type: ADD_NOTE,
    payload: {
      noteId,
      videoId,
      noteData: note
    }
  };
}

export function removeNote(noteId: NoteType['id'], videoId: VideoType['id']) {
  return {
    type: REMOVE_NOTE,
    payload: {
      noteId,
      videoId
    }
  };
}
