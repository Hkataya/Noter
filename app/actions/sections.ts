import { SectionType, CourseType } from '../reducers/entities/types';
import { Dispatch } from '../reducers/types';
import SectionRepository from '../db/SectionRepository';

export const ADD_SECTION = 'ADD_SECTION';
export const REMOVE_SECTION = 'REMOVE_SECTION';
export const FETCH_SECTIONS_BY_COURSE = 'FETCH_SECTIONS_BY_COURSE';
export const UPDATE_SECTION = 'UPDATE_SECTION';
type addSectionAction = {
  type: typeof ADD_SECTION;
  payload: {
    courseId: CourseType['id'];
    sectionId: SectionType['id'];
    sectionData: SectionType;
  };
};
type updateSection = {
  type: typeof UPDATE_SECTION;
  payload: {
    sectionId: SectionType['id'];
    sectionData: SectionType;
  };
};
type removeSectionAction = {
  type: typeof REMOVE_SECTION;
  payload: {
    sectionId: SectionType['id'];
  };
};

type FetchSectionsByCourseAction = {
  type: typeof FETCH_SECTIONS_BY_COURSE;
  payload: {
    sections: Array<SectionType>;
  };
};

export type SectionActionCreatorType = {
  addSectionDb?: (sectionData: Omit<SectionType, 'id'>) => unknown;
  removeSectionDb?: (sectionId: SectionType['id']) => unknown;
  fetchSectionsByCourseDb?: (courseId: CourseType['id']) => unknown;
  updateSection?: (sectionId: SectionType['id']) => unknown;
};

export type SectionActionType =
  | addSectionAction
  | removeSectionAction
  | FetchSectionsByCourseAction
  | updateSection;

export function addSection(sectionData: SectionType) {
  const sectionId = sectionData.id;
  const section = {
    ...sectionData
  };
  return {
    type: ADD_SECTION,
    payload: {
      sectionId,
      sectionData: section
    }
  };
}

export function removeSection(sectionId: SectionType['id']) {
  return {
    type: REMOVE_SECTION,
    payload: {
      sectionId
    }
  };
}
export function updateSection(
  sectionId: SectionType['id'],
  sectionData: SectionType
) {
  return {
    type: UPDATE_SECTION,
    payload: {
      sectionData
    }
  };
}
export function fetchSectionsByCourse(sectionsData: Array<SectionType>) {
  return {
    type: FETCH_SECTIONS_BY_COURSE,
    payload: {
      sections: sectionsData
    }
  };
}

export function addSectionDb(sectionData: Omit<SectionType, 'id'>) {
  return (dispatch: Dispatch) => {
    SectionRepository.createEntity(sectionData)
      .then(updatedData =>
        dispatch(addSection(Object.assign(sectionData, updatedData)))
      )
      .catch(err => {
        console.log(err);
      });
  };
}

export function removeSectionDb(sectionId: SectionType['id']) {
  return (dispatch: Dispatch) => {
    SectionRepository.deleteEntity(sectionId)
      .then(() => dispatch(removeSection(sectionId)))
      .catch(err => {
        console.log(err);
      });
  };
}

export function fetchSectionsByCourseDb(courseId: CourseType['id']) {
  return (dispatch: Dispatch) => {
    if (courseId) {
      SectionRepository.getSectionsByCourseId(courseId)
        .then((sections: Array<SectionType>) => {
          return dispatch(fetchSectionsByCourse(sections));
        })
        .catch(err => {
          console.log(err);
        });
    }
  };
}
export function updatedDataDb(sectionData: NonNullable<SectionType>) {
  console.log('updating note', sectionData);
  return (dispatch: Dispatch) => {
    SectionRepository.updateEntity(sectionData)
      .then(updatedData => dispatch(updateSection(sectionData.id, sectionData)))
      .catch((err: any) => {
        console.log(err);
      });
  };
}
