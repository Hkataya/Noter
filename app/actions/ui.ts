import { ModalType } from '../reducers/ui/types';
import { NoteType } from '../reducers/entities/types';

export const OPEN_MODAL = 'OPEN_MODAL';
export const CLOSE_MODAL = 'CLOSE_MODAL';
export const SET_CURRENTLY_SELECTED = 'SET_CURRENTLY_SELECTED';
export const SET_CURRENT_TIMESTAMP = 'SET_CURRENT_TIMESTAMP';
export const SET_TARGET_TIMESTAMP = 'SET_TARGET_TIMESTAMP';

type SetCurrentlySelected = {
  type: typeof SET_CURRENTLY_SELECTED;
  payload: string;
};

type OpenModalAction = {
  type: typeof OPEN_MODAL;
  payload: ModalType;
};

type CloseModalAction = {
  type: typeof CLOSE_MODAL;
};

type SetCurrentTimestampAction = {
  type: typeof SET_CURRENT_TIMESTAMP;
  payload: NoteType['timestamp'];
};

type SetTargetTimestampAction = {
  type: typeof SET_TARGET_TIMESTAMP;
  payload: NoteType['timestamp'];
};

export type UIActionCreatorType = {
  openModal?: (
    data: ModalType['data'],
    parentId?: ModalType['parentId'],
    type?: ModalType['type']
  ) => void;
  closeModal?: () => void;
  setCurrentlySelected?: (id: string) => void;
  setCurrentTimestamp?: (time: NoteType['timestamp']) => void;
  setTargetTimestamp?: (time: NoteType['timestamp']) => void;
};

export type UIActionType =
  | OpenModalAction
  | CloseModalAction
  | SetCurrentlySelected
  | SetCurrentTimestampAction
  | SetTargetTimestampAction;

export function openModal(
  data: OpenModalAction['payload']['data'],
  parentId: OpenModalAction['payload']['parentId'],
  type: OpenModalAction['payload']['type']
) {
  return {
    type: OPEN_MODAL,
    payload: {
      data,
      parentId,
      type
    }
  };
}

export function closeModal() {
  return {
    type: CLOSE_MODAL
  };
}

export function setCurrentlySelected(id: string) {
  return {
    type: SET_CURRENTLY_SELECTED,
    payload: id
  };
}

export function setCurrentTimestamp(time: NoteType['timestamp']) {
  return {
    type: SET_CURRENT_TIMESTAMP,
    payload: time
  };
}

export function setTargetTimestamp(time: NoteType['timestamp']) {
  return {
    type: SET_TARGET_TIMESTAMP,
    payload: time
  };
}
