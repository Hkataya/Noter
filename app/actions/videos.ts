import { VideoType, SectionType } from '../reducers/entities/types';
import { Dispatch } from '../reducers/types';
import { videoRepository as VideoRepository } from '../db/RepositoryInitializer';

export const ADD_VIDEO = 'ADD_VIDEO';
export const REMOVE_VIDEO = 'REMOVE_VIDEO';
export const TOGGLE_WATCHED = 'TOGGLE_WATCHED';
export const FETCH_VIDEOS_BY_SECTION = 'FETCH_VIDEOS_BY_SECTION';
export const UPDATE_VIDEO = 'UPDATE_VIDEO';
type UpdateVideoAction = {
  type: typeof UPDATE_VIDEO;
  payload: {
    videoData: VideoType;
  };
};
type AddVideoAction = {
  type: typeof ADD_VIDEO;
  payload: {
    videoId: VideoType['id'];
    videoData: VideoType;
  };
};

type RemoveVideoAction = {
  type: typeof REMOVE_VIDEO;
  payload: {
    videoId: VideoType['id'];
  };
};

type ToggleWatchedAction = {
  type: typeof TOGGLE_WATCHED;
  payload: {
    videoId: VideoType['id'];
  };
};

type FetchVideosBySectionAction = {
  type: typeof FETCH_VIDEOS_BY_SECTION;
  payload: {
    videos: Array<VideoType>;
  };
};

export type VideoActionCreatorType = {
  addVideoDb?: (videoData: Omit<VideoType, 'id'>) => unknown;
  removeVideoDb?: (videoId: VideoType['id']) => unknown;
  toggleWatched?: (videoId: VideoType['id']) => void;
  fetchVideosBySectionDb?: (sectionId: SectionType['id']) => unknown;
  updateVideoDb?: (videoData: VideoType) => unknown;
};

export type VideoActionType =
  | AddVideoAction
  | RemoveVideoAction
  | ToggleWatchedAction
  | FetchVideosBySectionAction
  | UpdateVideoAction;

export function addVideo(videoData: VideoType) {
  const videoId = videoData.id;
  return {
    type: ADD_VIDEO,
    payload: {
      videoId,
      videoData
    }
  };
}

export function removeVideo(videoId: VideoType['id']) {
  return {
    type: REMOVE_VIDEO,
    payload: {
      videoId
    }
  };
}
export function updateVideo(videoData: VideoType) {
  return {
    type: UPDATE_VIDEO,
    payload: {
      videoData
    }
  };
}
export function toggleWatched(videoId: VideoType['id']) {
  return {
    type: TOGGLE_WATCHED,
    payload: {
      videoId
    }
  };
}

export function fetchVideosBySection(videosData: Array<VideoType>) {
  return {
    type: FETCH_VIDEOS_BY_SECTION,
    payload: {
      videos: videosData
    }
  };
}

export function addVideoDb(videoData: Omit<VideoType, 'id'>) {
  return (dispatch: Dispatch) => {
    VideoRepository.createEntity(videoData)
      .then(updatedData =>
        dispatch(addVideo(Object.assign(videoData, updatedData)))
      )
      .catch(err => {
        console.log(err);
      });
  };
}

export function removeVideoDb(videoId: VideoType['id']) {
  return (dispatch: Dispatch) => {
    VideoRepository.deleteEntity(videoId)
      .then(() => dispatch(removeVideo(videoId)))
      .catch(err => {
        console.log(err);
      });
  };
}

export function fetchVideosBySectionDb(sectionId: SectionType['id']) {
  return (dispatch: Dispatch) => {
    if (sectionId) {
      VideoRepository.getVideosBySectionId(sectionId)
        .then((videos: Array<VideoType>) => {
          return dispatch(fetchVideosBySection(videos));
        })
        .catch(err => {
          console.log(err);
        });
    }
  };
}
export function updateVideoDb(videoData: VideoType) {
  return (dispatch: Dispatch) => {
    VideoRepository.updateEntity(videoData)
      .then(updatedData =>
        dispatch(updateVideo(Object.assign(videoData, updatedData)))
      )
      .catch(err => {
        console.log(err);
      });
  };
}
