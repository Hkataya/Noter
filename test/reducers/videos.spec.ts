import videos from '../../app/reducers/entities/videos';
import { ADD_VIDEO, REMOVE_VIDEO } from '../../app/actions/videos';

jest.mock('../../app/db/RepositoryInitializer', () => ({
  __esModule: true,
  default: 'mockedDefaultExport',
  namedExport: jest.fn()
}));

describe('videos', () => {
  it('should handle ADD_VIDEO', () => {
    expect(
      videos(
        {},
        {
          type: ADD_VIDEO,
          payload: {
            videoData: {
              title: 'test',
              id: '456',
              url: 'testurl',
              watched: false,
              section: '123'
            },
            videoId: '456'
          }
        }
      )
    ).toMatchSnapshot();
  });

  it('should handle REMOVE_VIDEO', () => {
    expect(
      videos(
        {
          '456': {
            title: 'test',
            id: '456',
            url: 'testurl',
            watched: false,
            section: '123'
          }
        },
        {
          type: REMOVE_VIDEO,
          payload: { videoId: '456' }
        }
      )
    ).toMatchSnapshot();
  });
});
