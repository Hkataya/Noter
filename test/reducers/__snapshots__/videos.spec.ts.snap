// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`videos should handle ADD_VIDEO 1`] = `
Object {
  "456": Object {
    "id": "456",
    "section": "123",
    "title": "test",
    "url": "testurl",
    "watched": false,
  },
}
`;

exports[`videos should handle REMOVE_VIDEO 1`] = `Object {}`;
